packages/annotorious/vite.config.js

```diff
@@ -21,14 +17,20 @@ export default defineConfig({
       entry: './src/index.ts',
       name: 'Annotorious',
       formats: ['es', 'umd'],
-      fileName: (format) => 
-        format === 'umd' ? `annotorious.js` : `annotorious.es.js` 
+      fileName: (format) =>
+        format === 'umd' ? `annotorious.js` : `annotorious.es.js`
     },
     rollupOptions: {
+      external: [
+        '@annotorious/core',
+        'test/*'
+      ],
```

Comment:
But here, again, I think this would break the UMD bundle.

Reply:
_It might... 🤔_ 
That's why I [asked](https://github.com/annotorious/annotorious/pull/388/files/ee552bc3a77ed5ef30e9fb631a509b0773593196#r1537904480) about the places where/whether the variables for the `globals` definitions are set in the consuming apps/packages. 
Because previously the `annotorious/annotorious` was already excluded from the bundling, being part of the `peerDependecies`. But it was being replaced with the `Annotorious` global variable. So, I suppose, this variable must have been set somewhere down the line 🤔

Reply:
I'm dying from the italicized emojis _🤣_ 
![image](https://github.com/annotorious/annotorious/assets/68850090/84739da2-35f8-49b0-a95d-40b05e6f5d08)

Reply:
But from [this answer](https://github.com/annotorious/annotorious/pull/388/files/ee552bc3a77ed5ef30e9fb631a509b0773593196#r1539111493), I surmise that the `Annotorious` was never set 🤷🏻‍♂️ 

**_So we might need to bundle the dependencies for the UDM compliance_**

Reply:
Not sure I understand the question. If you build the UMD library, Vite will create a global `Annotorious` variable, with all the properties you would otherwise import as named imports. I.e. the following code with the ES bundle:

```js
import { createImageAnnotator } from '@annotorious/annotorious';

const anno = createImageAnnotator(...);
```

will become this...

```js
var anno = Annotorious.createImageAnnotator(...);
```

...when you use the script import. Therefore, if @annotorious/core is not bundled into the UMD module as a direct depenendency, the UMD bundle will fail. Same for the AnnotoriousOSD UMD bundle, which needs to have both `@annotorious/core` and `@annotorious/annotorious` bundled in to work.